plugins/outputs/file/file.go

```diff
@@ -13,24 +13,64 @@ import (
 	"github.com/influxdata/telegraf/internal/rotate"
 	"github.com/influxdata/telegraf/plugins/outputs"
 	"github.com/influxdata/telegraf/plugins/serializers"
+	"github.com/klauspost/compress/zstd"
 )
 
 //go:embed sample.conf
 var sampleConfig string
 
+var ValidCompressionAlgorithmLevels = map[string][]int{
+	"zstd": {1, 3, 7, 11},
+}
+
+const (
+	defaultCompressionAlgorithm = "zstd"
+	defaultCompressionLevel     = 3
+)
+
 type File struct {
 	Files               []string        `toml:"files"`
 	RotationInterval    config.Duration `toml:"rotation_interval"`
 	RotationMaxSize     config.Size     `toml:"rotation_max_size"`
 	RotationMaxArchives int             `toml:"rotation_max_archives"`
 	UseBatchFormat      bool            `toml:"use_batch_format"`
+	Compression         Compression     `toml:"compression"`
+	Encoder             interface{}
```

Comment:
```suggestion
	encoder             interface{}
```
No need to export this.

Reply:
Agreed. it's been a while since I coded in go I forgot about this.